ai/matching/service.py

```
@@ -0,0 +1,322 @@
+        """
+        try:
+            alpha = alpha or self.alpha
+            beta = beta or self.beta
+
+            # Convert to numpy arrays
+            job_vec = np.array(job_vector).reshape(1, -1)
+            applicant_vec = np.array(applicant_vector).reshape(1, -1)
+
+            # Check dimension compatibility
+            if job_vec.shape[1] != applicant_vec.shape[1]:
+                raise ValueError(f"Vector dimension mismatch: job={job_vec.shape[1]}, applicant={applicant_vec.shape[1]}")
```

Ruff (0.13.2)

Abstract `raise` to an inner function

(TRY301)